record-and-playback/presentation/scripts/publish/presentation.rb

```diff
@@ -1225,6 +1228,15 @@ def processDeskshareEvents(events)
           BigBlueButton.logger.info("Copied audio.ogg file")
         end
 
+        if File.exist?("#{captions_meeting_dir}/captions.json")
```

Comment:
The captions are already being generated or copied in the `process` script. There shouldn't be any change required for the `publish` script.

Reply:
But captions are not generated for all recordings, right?

Reply:
I'm not sure what you're asking?

The caption files are copied into the recording directory during the `process` script. The `publish` script doesn't have to do anything regarding captions because the process script has already done it.